pf9/cluster/cluster_attach.py

```diff
@@ -105,14 +105,14 @@ def get_resmgr_hostid(self, host_ip):
 
         # sequentially search results
         for host in json_response:
-            if 'extensions' not in host:
-                continue
-            # TODO: Why is there a loop in a loop where the values from the first loop aren't used?
-            for key, value in host['extensions']['interfaces']['data'].items():
+            try:
                 for iface_name, iface_ip in host['extensions']['interfaces']['data']['iface_ip'].items():
                     if iface_ip == host_ip:
                         logger.info("Node host_id: {}".format(host['id']))
                         return host['id']
+            except Exception as e:
+                logger.exception(e)
```

Comment:
Return non-zero CLI exit code?

Reply:
There is no exit code handling in the code base. This needs a larger change. I suggest we take this up later if need be. The exception is still logged to the log file. That will provide debug information.